hawc/apps/invitro/exports.py

```diff
@@ -1,362 +1,622 @@
-
-            doses = [eg["dose"] for eg in ser["groups"]]
-            diffs = [eg["difference_control"] for eg in ser["groups"]]
-            sigs = [eg["significant_control"] for eg in ser["groups"]]
-            cytotoxes = [eg["cytotoxicity_observed"] for eg in ser["groups"]]
-
-            if doses and doses[0] == 0:
-                doses.pop(0)
-                diffs.pop(0)
-                sigs.pop(0)
-
-            number_doses = len(doses)
-
-            bm_types = [bm["benchmark"] for bm in ser["benchmarks"]]
-            bm_values = [bm["value"] for bm in ser["benchmarks"]]
-
-            study_id = ser["experiment"]["study"]["id"]
-            row = [
-                study_id,
-                identifiers_df["hero_id"].get(study_id),
-                identifiers_df["pubmed_id"].get(study_id),
-                identifiers_df["doi"].get(study_id),
-                ser["experiment"]["study"]["short_citation"],
-                ser["experiment"]["study"]["study_identifier"],
-                ser["experiment"]["study"]["published"],
-            ]
-
-            study_robs = [
-                self.rob_data[(study_id, metric_id)] for metric_id in self.rob_headers.keys()
-            ]
-            row.extend(study_robs)
-
-            row.extend(
-                [
-                    ser["chemical"]["id"],
-                    ser["chemical"]["name"],
-                    ser["chemical"]["cas"],
-                    ser["chemical"]["dtxsid"],
-                    ser["chemical"]["purity"],
-                    ser["experiment"]["id"],
-                    ser["experiment"]["cell_type"]["id"],
-                    ser["experiment"]["cell_type"]["species"],
-                    ser["experiment"]["cell_type"]["strain"],
-                    ser["experiment"]["cell_type"]["sex"],
-                    ser["experiment"]["cell_type"]["cell_type"],
-                    ser["experiment"]["cell_type"]["tissue"],
-                    ser["experiment"]["dose_units"]["name"],
-                    ser["experiment"]["metabolic_activation"],
-                    ser["experiment"]["transfection"],
-                    ser["id"],  # repeat for data-pivot key
-                    ser["id"],
-                    ser["name"],
-                    "|".join([d["name"] for d in ser["effects"]]),
-                    ser["assay_type"],
-                    ser["short_description"],
-                    ser["response_units"],
-                    ser["observation_time"],
-                    ser["observation_time_units"],
-                    getDose(ser, "NOEL"),
-                    getDose(ser, "LOEL"),
-                    ser["monotonicity"],
-                    ser["overall_pattern"],
-                    ser["trend_test"],
-                    doseRange[0],
-                    doseRange[1],
-                    number_doses,
+    def handle_benchmarks(self, df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
A similar approach to EndpointGroup could probably be applied here. Again, in the future, once we've got our comparisons scripts complete.